### ISSUE TITLE
Creating a Wikidata item with same label and description as another item creates a blank item

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#1859]
**To Reproduce**
1. Mark a cell as new and add a label and description in the schema
2. If the label and description are identical to those of another item, creating the item will fail to add these terms (the item will be empty)

**Expected behavior**
Refuse to create the item? How should we report this error to the user?
More broadly, there should be a way to report any issue encountered during editing. Add a new column and store the errors there?

**OpenRefine (please complete the following information):**
 - Version 3.0

**Comment by ostephens** [comment]
I think you are correct - the broader issue is how to report back errors to the user. 

An alternative to storing the error in a new column might be to add it to the reconciliation object. Since it might(?) be possible to have multiple errors for each row from a Upload Edits to Wikidata errors would need to be stored in a list or array. Making this part of the Recon object would mean you could see which errors referred to which item.

Possibly the Upload Edits to Wikidata could be followed by automatically creating a 'Reconciliation errors' facet on each column used as an 'Item' in the current schema?

In this particular scenario I think the right thing to do is probably to not create the item (because that's what happens if you try to directly create an item with duplicate label+description on WD)

**Comment by ostephens** [comment]
Adding a dedicated field to `com.google.refine.model.Recon` was what I was suggesting - which wouldn't necessarily be Wikidata specific - it could be used for storing errors from any reconciliation process?

At an even more general level it might make sense to have an error field on a Cell object. At the moment you have to choose between storing the error in the cell and throwing it away - which is a pain in some situations - I'm thinking especially of 'add column from URL' where if you forget to check 'store error' and get back a load of empty cells you have to do the process again to actually  find what the error was.

But just throwing ideas around really - I don't have strong views on this

**Comment by wetneb** [comment]
Yeah adding a dedicated field, either to `Recon` or to `Cell`, can potentially be useful indeed.

Another problem with storing errors in a dedicated column is what to do when there are no errors - for consistency I guess an empty column should be created (so that subsequent operations in a workflow can rely on it) but that is not very user-friendly. Letting the user choose in advance if they want to store errors is not an option either, for the reason you mention.

**Comment by thadguidry** [comment]
So is the consensus between the two of you to:
- add an Error field(s) to `com.google.refine.model.Recon`
- use customized Facet (for now, which can be moved to a future Recon panel, See Issue #1520 [(Thad's design comment)](https://github.com/OpenRefine/OpenRefine/issues/1520#issuecomment-368695297) ) to expose Recon Error fields
- Anything else ?

**Comment by salgo60** [comment]
My vote for this error field to. I did an upload yesterday and got no feedback of an error. 

I just found that out when starting matching and after that I tried a Quick statement with missing records and got an error message - that I had duplicated labels...

My ref: [T215603#4964368](https://phabricator.wikimedia.org/T215603#4964368)

**Comment by wetneb** [comment]
We could also store errors in a custom overlay model (different from the schema). That would give us more freedom and would not mess up with the core reconciliation data model. The errors stored there could be made available via a GREL function (given a Qid, return the editing error for this Qid, if any) and therefore facets.

**Comment by thadguidry** [comment]
@wetneb I like the idea of separation of concerns.  The new UI work could offer a view of those Errors and even faceting on them.  Multi-datagrids per project similar to Excel sheets, and having a split bar between the 2 sheets, ReconData|Errors, so you can Facet and view both sides at the same time.
